OpenRA.Mods.Common/Warheads/CreateResourceWarhead.cs

```diff
@@ -28,13 +28,16 @@ public class CreateResourceWarhead : Warhead
 		// TODO: Allow maximum resource splatter to be defined. (Per tile, and in total).
 		public override void DoImpact(Target target, WarheadArgs args)
 		{
+			var firedBy = args.SourceActor;
+			if (!target.IsValidFor(firedBy))
```

Comment:
This feels too specific, IMO. `target.Type != TargetType.Invalid` is all we need to know the `CenterPosition` is valid.